packages/leancode_contracts_generator/lib/src/contracts_generator.dart

```diff
@@ -39,8 +41,17 @@ class ContractsGenerator {
 
   /// generates `code_builder` structures that can still be modified
   Future<Library> generate() async {
+    final doneDb = verboseLog(
+      message: 'Analysing contracts',
+      verbose: verbose,
+    );
     final db = await GeneratorDatabase.fromConfig(config);
+    doneDb();
```

Comment:
You might want to use [package:mason_logger](https://pub.dev/packages/mason_logger), which has a cool [Progress](https://pub.dev/documentation/mason_logger/latest/mason_logger/Progress-class.html) class.

Reply:
I totally get you. Cool choice 👍

> inspired by Golang

I'd say inspired [by Unix instead](https://www.arp242.net/the-art-of-unix-programming/#id2878450)

Reply:
Oh, thanks! Had no idea its from Unix, good to know